### PR TITLE
[Infra] Add issue form for bug reports

### PR DESCRIPTION
**Pull request by ncooke3**
### Context
[GitHub issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) offer a better way to format and template GitHub issues.

I have ported the markdown issue template to the issue form syntax. I also added some fields that I think would be helpful for triage.

I have a fork where you can see it in action. 

See [here](https://github.com/ncooke3/firebase-ios-sdk/issues/new?assignees=&labels=&template=BUG_REPORT.yml) for what it looks like when opening an issue. 

And see [here](https://github.com/ncooke3/firebase-ios-sdk/issues/2) for what the created issue looks like.

I played around with the format and ordering and am open to suggestions!

#no-changelog

### CODE REVIEW

**Comment by paulb777** [comment]
Looks great!  Is there a way to make the Podfile.lock unexpanded by default?

While it is often very useful, I'm worried that the length could be too much sometimes.

**Comment by ncooke3** [comment]
@paulb777, I had to get a little creative but check it out:

- [What you see when opening issue](https://github.com/ncooke3/firebase-ios-sdk/issues/new?assignees=&labels=&template=BUG_REPORT.yml)
- [What the created issue looks like](https://github.com/ncooke3/firebase-ios-sdk/issues/4)

**Comment by paulb777** [comment]
Great! I like it.

Only a minor nit of being more descriptive than "inside the code block". How about after the line with `yml` in it?

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        involves code in this repository not working as expected, please
+        complete and submit this bug report. Thank you!
+  - type: textarea
+    attributes:
+      label: Description
+      description: |
+        Describe the issue. Explain what you _expected_ to happen and what
+        _actually_ happened.
+    validations:
+      required: true
```
Comment (ncooke3) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:28]: One thing to note for reviewers, the [fields marked with the red asterisks are required to submit the form](https://github.com/ncooke3/firebase-ios-sdk/issues/new?assignees=&labels=&template=BUG_REPORT.yml). I put some thought into whether a field should be required or not but LMK what you think.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      label: If using CocoaPods, the project's Podfile.lock
+      description: The `Podfile.lock` can help us debug versioning issues.
+      value: |
+        <!--- Look below for instructions on how to share your Podfile.lock. --->
+
+        <details>
+        <summary>Expand <code>Podfile.lock</code> snippet</summary>
+        <br>
+
+        <!--- Paste the contents of your Podfile.lock *inside* the code block below.
```
Comment (paulb777) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Posters may not know how to translate "code block" to what's below.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        _actually_ happened.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Reproducing the issue
+      description: |
+        Please provide either **steps to reproduce** or a [**minimal reproducible example**](https://stackoverflow.com/help/minimal-reproducible-example).
+        Providing a minimal reproducible example will help us triage your issue
+        much faster.
```
Comment (paulb777) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:35]: nit: delete "much"

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 2 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Resolved in [2a10e3d](https://github.com/firebase/firebase-ios-sdk/pull/10119/commits/2a10e3d08c4bb272ea14970402d5a804d3237b90).

LMK if you think it needs more work.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 3 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Done in bed90f8

**Review by paulb777 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        - tvOS
+        - watchOS
+        - macOS
+        - macCatalyst
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Relevant Log Output
+      description: Please copy and paste any relevant log output.
```
[the remark this answers is not in the document]
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:104]: Great suggestion!

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        - Analytics
+        - App Check
+        - App Distribution
+        - Authentication
+        - Crashlytics
+        - Database
+        - DynamicLinks
+        - Firestore
+        - Functions
+        - In App Messaging
```
Comment (morganchen12) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:79]: nit: `In-App`

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    attributes:
+      label: Targeted Platforms
+      description: What Apple platform(s) is the issue affecting?
+      multiple: true
+      options:
+        - iOS
+        - tvOS
+        - watchOS
+        - macOS
+        - macCatalyst
```
Comment (morganchen12) [inline .github/ISSUE_TEMPLATE/BUG_REPORT.yml:98]: maybe add extensions?

**Review by morganchen12 — APPROVED** [review]

**Review by rizafran — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Relevant Log Output
+      description: Please copy and paste any relevant log output.
+      placeholder: |
+        Paste the your logs here. Please redact any personally identifiable
```
[the remark this answers is not in the document]
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:106]: good find! Done.

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Done in 7c34f5b

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 5 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml]: Done in [7c34f5b](https://github.com/firebase/firebase-ios-sdk/pull/10119/commits/7c34f5b3fa726ec7bbcb6930b181a3ef04a22f00)

File: .github/ISSUE_TEMPLATE/BUG_REPORT.yml (hunk 6 above)
Comment (ncooke3) [reply .github/ISSUE_TEMPLATE/BUG_REPORT.yml:98]: Done in 7c34f5b

**Pull request merged**